components/brave_wallet/browser/brave_wallet_constants.cc

```diff
@@ -16,22 +15,21 @@
 
 namespace brave_wallet {
 
-const std::string GetSardineNetworkName(const std::string& chain_id) {
+std::optional<std::string> GetSardineNetworkName(std::string_view chain_id) {
   // key = chain_id, value = sardine_network_name
-  static std::map<std::string, std::string> sardine_network_names = {
-      {mojom::kMainnetChainId, "ethereum"},
-      {mojom::kPolygonMainnetChainId, "ethereum"},
-      {mojom::kAvalancheMainnetChainId, "avalanche"},
-      {mojom::kFantomMainnetChainId, "fantom"},
-      {mojom::kSolanaMainnet, "solana"}};
-  auto sardine_network_pair = sardine_network_names.find(chain_id.c_str());
-
-  if (sardine_network_pair == sardine_network_names.end()) {
+  static base::NoDestructor<base::flat_map<std::string, std::string>>
```

Comment:
fixed_flat_map?

Reply:
Unfortunately we can't use `fixed_flat_map` because it requires `constexpr` types, and `mojom::kMainnetChainId` and similar constants are `extern` values that are not `constexpr` friendly.